modules/apps/frontend-js/frontend-js-module-launcher-api/src/main/java/com/liferay/frontend/js/module/launcher/JSModuleLauncher.java

```diff
@@ -0,0 +1,53 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.frontend.js.module.launcher;
+
+import java.io.Writer;
+
+import java.util.Collection;
+
+import javax.servlet.http.HttpServletRequest;
+
+/**
+ * @author Iván Zaera Avellón
+ */
+public interface JSModuleLauncher {
+
+	public void appendPortletScript(
+		HttpServletRequest httpServletRequest, String portletId,
+		Collection<JSModuleDependency> jsModuleDependencies,
+		String javascriptCode);
+
+	public void appendScript(
+		HttpServletRequest httpServletRequest,
+		Collection<JSModuleDependency> jsModuleDependencies,
+		String javascriptCode);
+
+	/**
+	 * Check if a JavaScript module identified by name must be loaded using
+	 * {@link JSModuleLauncher} (instead of AMD Loader, for example).
+	 *
+	 * @review
+	 */
```

Comment:
Was this what @wincent was referring to in Slack? Why this one has jsdoc and not the others? 😂

Reply:
Because this one is very fortunate and the others are losers

![losers](https://media.giphy.com/media/3o7TKr3nzbh5WgCFxe/giphy.gif)